KalturaWowzaServer/src/main/java/com/kaltura/media/server/wowza/CuePointsManager.java

```diff
@@ -277,138 +276,146 @@ public void onLiveStreamPacketizerInit(ILiveStreamPacketizer liveStreamPacketize
 	@Override
 	public void init() throws KalturaManagerException {
 		super.init();
-
-		KalturaEventsManager.registerEventConsumer(this, KalturaEventType.STREAM_PUBLISHED, KalturaMediaEventType.APPLICATION_INSTANCE_STARTED);
+		KalturaEventsManager.registerEventConsumer(this, KalturaEventType.STREAM_PUBLISHED, KalturaEventType.STREAM_UNPUBLISHED, KalturaMediaEventType.APPLICATION_INSTANCE_STARTED);
+		liveManager = KalturaServer.getManager(ILiveManager.class);
 		setInitialized();
 	}
 
 	@Override
-	public void onEvent(IKalturaEvent event){
+	public void onEvent(IKalturaEvent event) {
 
-		if(event.getType() instanceof KalturaMediaEventType){
+		if (event.getType() instanceof KalturaMediaEventType) {
 
-			switch((KalturaMediaEventType) event.getType())
-			{
+			switch ((KalturaMediaEventType) event.getType()) {
 				case APPLICATION_INSTANCE_STARTED:
 					KalturaApplicationInstanceEvent applicationInstanceEvent = (KalturaApplicationInstanceEvent) event;
 					onAppStart(applicationInstanceEvent.getApplicationInstance());
 					break;
-
 				default:
 					break;
 			}
 		}
 
-		if(event.getType() instanceof KalturaEventType){
+		if (event.getType() instanceof KalturaEventType) {
 
-			switch((KalturaEventType) event.getType())
-			{
+			KalturaMediaStreamEvent streamEvent = (KalturaMediaStreamEvent) event;
```

Comment:
Please check - will this cast surely succeed? (It wasn't done for unpublish before)